modules/@apostrophecms/rich-text-widget/ui/apos/components/ApostropheTiptapLink.vue

```diff
@@ -1,89 +1,218 @@
-          </fieldset>
-        </form>
-      </template>
-      <template slot="footer">
-        <slot name="footer">
-          <button class="modal-default-button" @click="close">
-            Cancel
-          </button>
-          <button class="modal-default-button" @click="save()">
-            Save
-          </button>
-        </slot>
-      </template>
-    </ApostropheModal>
-  </div>
-</template>
-
-
+    <editor-menu-bubble
+      :editor="editor"
+      :keep-in-bounds="keepInBounds"
+      v-slot="{ menu }"
+    >
+      <div
+        class="apos-link-control__dialog"
+        :class="{
+          'is-ready': menu.isActive,
+          'is-triggered': active,
+          'has-selection': hasSelection
+        }"
+        :style="
+          `left: ${menu.left}px;
+          bottom: ${menu.bottom}px;
+          transform: translate3d(-25px, calc(100% + ${offset}), 0);
+          `"
+      >
+        <AposContextMenuDialog
+          menu-placement="bottom-start"
+          v-if="active"
+        >
+          <form>
+            <AposSchema
+              :schema="schema"
+              v-model="value"
+            />
+            <footer class="apos-link-control__footer">
+              <AposButton
+                type="default" label="Cancel"
+                @click="close"
+              />
+              <AposButton
+                type="primary" label="Save"
+                @click="save"
+              />
+            </footer>
+          </form>
+        </AposContextMenuDialog>
+      </div>
+    </editor-menu-bubble>
   </div>
 </template>
 
 <script>
 
+import { EditorMenuBubble } from 'tiptap';
+import { isEmpty } from 'lodash';
+import {
+  VPopover
+} from 'v-tooltip';
+
 export default {
   name: 'ApostropheTiptapLink',
+  components: {
+    'v-popover': VPopover,
+    EditorMenuBubble
+  },
   props: {
-    name: String,
-    tool: Object,
-    editor: Object
+    name: {
+      type: String,
+      required: true
+    },
+    tool: {
+      type: Object,
+      required: true
+    },
+    editor: {
+      type: Object,
+      required: true
+    }
   },
   data () {
     return {
-      href: '',
-      id: '',
-      target: '',
-      active: false
+      keepInBounds: true,
+      href: null,
+      id: null,
+      target: null,
+      active: false,
+      value: {
+        data: {}
+      },
+      schema: [
+        {
+          name: 'href',
+          label: 'URL',
+          help: 'Where should the link go?',
+          placeholder: 'http://calm.com',
+          type: 'string'
+        },
+        {
+          name: 'id',
+          label: 'Anchor Name',
+          help: 'This becomes the ID of the anchor',
+          type: 'string'
+        },
+        {
+          name: 'target',
+          label: 'Target',
+          help: 'Where should this link open?',
+          type: 'select',
+          def: '_self',
+          choices: [
+            {
+              label: 'Current browsing context (_self)',
```

Comment:
Minor, but can we just simplify this a bit? Can we just have "Current Tab" and "New Tab"? Its a bit technical as written

Reply:
Yeah open to better labels on these, i think i either c+p'd mdn or paraphrased

Reply:
updated this